### ISSUE TITLE
re-evaluate resource requirements for building Kubernetes from source

### ISSUE THREAD

**Issue by pohly** [kubernetes/test-infra#28800]
**What should be cleaned up or changed**:

Once upon a time, someone (`@bentheelder`?) determined how much memory was needed to build Kubernetes from source: `memory: "9000Mi"`.

This may or may not be correct nowadays. We should check (how?) and then update all jobs using this magic value accordingly. Ideally it's lower and the shared cluster for running these jobs then can be used more efficiently.

**Provide any links for context**:

https://grep.app/search?q=memory%3A%20%229000Mi%22

**Comment by BenTheElder** [comment]
https://monitoring-eks.prow.k8s.io/
https://monitoring-gke.prow.k8s.io/

I would hazard we're able to build at 8 cores x 5Gi currently, when building for a single architecture, in a stripped down build (note kind does not do a full quick release) https://monitoring-gke.prow.k8s.io/d/96Q8oOOZk/builds?var-org=kubernetes&var-repo=kubernetes&var-job=pull-kubernetes-e2e-kind&orgId=1&refresh=30s

We could look at more jobs, I will try to circle back to this at some point.

/sig testing release k8s-infra

**Comment by anshikavashistha** [comment]
@pohly Is this issue still open to work ?

**Comment by pohly** [comment]
Yes. The importance however is low: our CI is currently constrained more by cores than memory.

**Comment by anshikavashistha** [comment]
> Yes. The importance however is low: our CI is currently constrained more by cores than memory.

@pohly What tooling or approach would you recommend for measuring the memory usage during a build?

**Comment by pohly** [comment]
We have dashboards: https://github.com/kubernetes/test-infra/issues/34139#issuecomment-2593811973
